tests/unit/conftest.py

```diff
@@ -99,3 +100,46 @@ def ingress_requirer_application_data_fixture() -> dict[str, str]:
 def ingress_requirer_unit_data_fixture() -> dict[str, str]:
     """Mock ingress requirer unit data."""
     return {"host": '"testing.ingress"', "ip": '"10.0.0.1"'}
+
+
+# Scenario
+@pytest.fixture(name="context_with_install_mock")
+def context_with_install_mock_fixture():
+    """Context relation fixture.
+
+    Yield: The modeled haproxy-peers relation.
+    """
+    with patch("haproxy.HAProxyService.install") as install_mock:
+        yield (
+            Context(
+                charm_type=HAProxyCharm,
+            ),
+            install_mock,
+        )
+
+
+@pytest.fixture(name="peer_relation")
+def peer_relation_fixture():
+    """Peer relation fixture.
+
+    Yield: The modeled haproxy-peers relation.
+    """
+    yield scenario.PeerRelation(
+        endpoint="haproxy-peers",
+        peers_data={},
+    )
```

Comment:
Nit: I think if we're not running cleanup code, we can use `return`
```suggestion
    return scenario.PeerRelation(
        endpoint="haproxy-peers",
        peers_data={},
    )
```

Reply:
same for other functions but up to you!